globus_automate_client/models.py

```diff
@@ -0,0 +1,392 @@
+    def validate_mutually_exclusive_fields(cls, values):
+        fields = ["InputPath", "Parameters"]
+        set_fields = {f: values.get(f) for f in fields if values.get(f) is not None}
+        if len(set_fields) > 1:
+            raise ValueError(
+                f"Only one of these fields may be set: {set_fields.keys()}"
+            )
+        if len(set_fields) == 0:
+            raise ValueError(f"At least one field must be set: {fields}")
+
+        return values
+
+
+class ExpressionEvalState(NextOrEndState):
+    Type: te.Literal[StateType.ExpressionEval]
+    Comment: t.Optional[str] = None
+    Parameters: t.Optional[dict] = None
+    ResultPath: t.Optional[str] = None
+
+    class Config:
+        extra = Extra.forbid
+
+    _enforce_jsonpath = validator("ResultPath", allow_reuse=True)(enforce_jsonpath)
+    _check_jsonpath_syntax_in_dict = validator("Parameters", allow_reuse=True)(
+        check_jsonpath_syntax_in_dict
+    )
+
+
+class FailState(BaseModel):
+    Type: te.Literal[StateType.Fail]
+    Comment: t.Optional[str] = None
+    Cause: t.Optional[str] = None
+    Error: t.Optional[str] = None
+
+    class Config:
+        extra = Extra.forbid
+
+
+class PassState(NextOrEndState):
+    Type: te.Literal[StateType.Pass]
+    Comment: t.Optional[str] = None
+    Parameters: t.Optional[t.Dict[str, t.Any]] = None
+    InputPath: t.Optional[str] = None
+    Result: t.Optional[dict] = None
+    ResultPath: t.Optional[str] = None
+    # Not supported by Flows
+    # OutputPath: t.Optional[str] = None
+
+    class Config:
+        extra = Extra.forbid
+
+    _enforce_jsonpath = validator(
+        "InputPath", "Result", "ResultPath", allow_reuse=True
+    )(enforce_jsonpath)
+    _check_jsonpath_syntax_in_dict = validator("Parameters", allow_reuse=True)(
+        check_jsonpath_syntax_in_dict
+    )
+
+
+class WaitState(NextOrEndState):
+    Type: te.Literal[StateType.Wait]
+    Comment: t.Optional[str] = None
+    InputPath: t.Optional[str] = None
+    OutputPath: t.Optional[str] = None
+    Seconds: t.Optional[int] = None
+    Timestamp: t.Optional[datetime.datetime] = None
+    SecondsPath: t.Optional[str] = None
+    TimestampPath: t.Optional[str] = None
+
+    _enforce_jsonpath = validator(
+        "InputPath", "OutputPath", "SecondsPath", "TimestampPath", allow_reuse=True
+    )(enforce_jsonpath)
+
+    class Config:
+        extra = Extra.forbid
+
+    @root_validator(skip_on_failure=True)
+    def validate_mutually_exclusive_fields(cls, values):
+        fields = ["Seconds", "Timestamp", "SecondsPath", "TimestampPath"]
+        set_fields = {f: values.get(f) for f in fields if values.get(f) is not None}
+        if len(set_fields) > 1:
+            raise ValueError(
+                f"Only one of these fields may be set: {set_fields.keys()}"
+            )
+        if len(set_fields) == 0:
+            raise ValueError(f"At least one field must be set: {fields}")
+
+        return values
+
+
+FlowState = te.Annotated[
```

Comment:
This is the clever/correct solution for setting up the discriminator on a type which is only used as a `TypeAnnotation` (in the `States` dictionary below). Nice One!